backend/src/main/routes/AccountRoute.ts

```diff
@@ -28,13 +28,24 @@ export default class AccountRoute extends CommonRoutesConfig {
   configureRoutes(): express.Application {
     this.getApp()
       .route(`/accounts/employee`)
-      .all(checkJwt, checkRole(new Set([Roles.BUSINESS])))
-      .post(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      .post(
+        checkJwt,
+        checkRole(new Set([Roles.BUSINESS])),
+        async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+          try {
+            const newEmployeeAccount = await this.employeeAccountService.createEmployeeAccount(req.body);
+            const dto = JSON.parse(JSON.stringify(newEmployeeAccount));
+            delete dto.account.password;
+            res.status(StatusCodes.CREATED).send(dto);
+          } catch (err) {
+            next(err);
+          }
+        }
+      )
+      .get(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
         try {
-          const newEmployeeAccount = await this.employeeAccountService.createEmployeeAccount(req.body);
-          const dto = JSON.parse(JSON.stringify(newEmployeeAccount));
-          delete dto.account.password;
-          res.status(StatusCodes.CREATED).send(dto);
+          const regexEmployeeAccount = await this.employeeAccountService.getEmployeesByRegex(String(req.query.email));
+          res.status(StatusCodes.OK).send(regexEmployeeAccount);
         } catch (err) {
           next(err);
         }
```

Comment:
For /accounts/allEmployees and /accounts/allEmployees/:business, I don't know why but it seems like it reverted the changes made in the main. This is what is causing the code smell. Could you try pulling and merging the changes from the main branch to your branch?
Main branch code:
![image](https://user-images.githubusercontent.com/48056716/147854164-41e88e82-64de-4ee3-ab13-5f2d257245a3.png)

Reply:
fixed

Reply:
It doesn't show up as changes so I won't revert back. It were to revert back, it would have shown the old code in the changes since it is comparing it to main. It's probably because, this branch hasn't been updated with main.